### PR TITLE
tracing: replace Jaeger with Sentry

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

**Comment by monstermunchkin** [comment]
We might want to merge `maintenance/errors` and `maintenance/tracing` into `maintenance/sentry` once the raven MR has been merged. Currently both packages need to initialize sentry which can lead to issues if they're not in sync.

File: http/transport/tracing_round_tripper.go
Comment (arnold-iakab) [inline http/transport/tracing_round_tripper.go:31]: Should this round tripper contain the "hub" logic here? Might be cases where this roundtripper is used on a client that we use from outside of a request flow, so I'm thinking if it would make sense to have that in place here or not. Or is the whole "hub" mechanism relevant for keeping track of requests that touch multiple services? 🤔

**Review by arnold-iakab — CHANGES_REQUESTED** [review]
LGTM overall. I assume we will not merge this until we use this branch in `go-webkit` and at least one service to see how it behaves after review deployed, right?

File: go.mod
Comment (srieckhof) [reply go.mod:162]: Well, it's an indirect dependency. It's obviously required by one of our dependencies from the first require section.

File: backend/postgres/postgres.go
Comment (srieckhof) [reply backend/postgres/postgres.go:19]: There is .golanci.yml in place now. It is very rudimentary still but you could add a linter that at least checks on this. Although: This potential new linter may spam this PR with changes, right?

File: http/jsonapi/generator/internal/articles/open-api_test.go
Comment (monstermunchkin) [reply http/jsonapi/generator/internal/articles/open-api_test.go:68]: That is correct.

File: maintenance/tracing/tracing.go
Comment (srieckhof) [inline maintenance/tracing/tracing.go:24]: Shouldn't that be `%v`?

File: maintenance/tracing/tracing.go
Comment (srieckhof) [inline maintenance/tracing/tracing.go:81]:
```suggestion
	span := sentry.SpanFromContext(ctx)
```
?

File: http/jsonapi/generator/generate_handler.go
Comment (monstermunchkin) [reply http/jsonapi/generator/generate_handler.go:615]: This generates the following code, depending on whether or not the context variable has previously been defined or not:
```go
		ctx := span.Context()
		r = r.WithContext(ctx)
```

or 

```go
		ctx = span.Context()
		r = r.WithContext(ctx)
```

I added these lines to ensure the ctx contains the span, and that the request contains that same context.

File: backend/redis/redis.go
Comment (monstermunchkin) [reply backend/redis/redis.go]: Done.

File: http/transport/tracing_round_tripper.go
Comment (monstermunchkin) [reply http/transport/tracing_round_tripper.go]: I've added the hub logic.

File: maintenance/tracing/tracing.go
Comment (monstermunchkin) [reply maintenance/tracing/tracing.go]: Done.

**Review by srieckhof — APPROVED** [review]

File: http/jsonapi/generator/generate_handler.go
Comment (srieckhof) [reply http/jsonapi/generator/generate_handler.go:615]: You probably meant to paste a different code snippet in the second example?

Comment (monstermunchkin) [reply http/jsonapi/generator/generate_handler.go:615]: No, the pasted code snippets are both correct.

File: http/oauth2/oauth2.go
Comment (srieckhof) [inline http/oauth2/oauth2.go:67]: What happened to the "OAuth2"?

**Review by arnold-iakab — APPROVED** [review]

File: http/oauth2/oauth2.go
Comment (monstermunchkin) [reply http/oauth2/oauth2.go:67]: I changed the span name (or description) to `introspectRequest` as that's what the function does.

**Review by srieckhof — APPROVED** [review]

**Pull request merged**
